From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 10 Aug 2025 15:44:13 +0200
Subject: environment: Mock more DesktopAppInfo methods

Commit ac113accf1e0927 didn't mock all DesktopAppInfo methods used by
gnome-shell, and https://gitlab.gnome.org/GNOME/glib/-/issues/3744
still hasn't been fixed.

Add the methods that are needed for the AppMenu, which hopefully are
all the remaining DesktopAppInfo methods we are using.

Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8587
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3832>
---
 js/ui/environment.js | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/js/ui/environment.js b/js/ui/environment.js
index bbbdf1d..a008a37 100644
--- a/js/ui/environment.js
+++ b/js/ui/environment.js
@@ -347,9 +347,21 @@ if (!GioUnix.DesktopAppInfo.prototype.has_key) {
         return GioUnix.DesktopAppInfo.has_key(this, key);
     };
 
+    GioUnix.DesktopAppInfo.prototype.get_boolean = function (key) {
+        return GioUnix.DesktopAppInfo.get_boolean(this, key);
+    };
+
     GioUnix.DesktopAppInfo.prototype.get_categories = function () {
         return GioUnix.DesktopAppInfo.get_categories(this);
     };
+
+    GioUnix.DesktopAppInfo.prototype.list_actions = function () {
+        return GioUnix.DesktopAppInfo.list_actions(this);
+    };
+
+    GioUnix.DesktopAppInfo.prototype.get_action_name = function (name) {
+        return GioUnix.DesktopAppInfo.get_action_name(this, name);
+    };
 }
 
 const slowdownEnv = GLib.getenv('GNOME_SHELL_SLOWDOWN_FACTOR');
